llama_index/response_synthesizers/google/generativeai/base.py

```diff
@@ -0,0 +1,131 @@
+"""Google GenerativeAI Attributed Question and Answering (AQA) service.
+
+The GenAI Semantic AQA API is a managed end to end service that allows
+developers to create responses grounded on specified passages based on
+a user query. For more information visit:
+https://developers.generativeai.google/guide
+"""
+
+import logging
+from typing import Any, List, Sequence, cast
+
+from llama_index.bridge.pydantic import BaseModel  # type: ignore
+from llama_index.prompts.mixin import PromptDictType
+from llama_index.response_synthesizers.base import BaseSynthesizer
+from llama_index.types import RESPONSE_TEXT_TYPE
+from llama_index.vector_stores.google.generativeai import google_service_context
+
+_logger = logging.getLogger(__name__)
+_import_err_msg = "`google.generativeai` package not found, please run `pip install google-generativeai`"
+_separator = "\n\n"
+
+
+class SynthesizedResponse(BaseModel):
+    """Response of `GoogleTextSynthesizer.get_response`."""
+
+    answer: str
+    """The grounded response to the user's question."""
+
+    attributed_passages: List[str]
+    """The list of passages the AQA model used for its response."""
+
+    answerable_probability: float
+    """The probability of the question being answered from the provided passages."""
+
+
+class GoogleTextSynthesizer(BaseSynthesizer):
+    """Google's Attributed Question and Answering service.
+
+    Given a user's query and a list of passages, Google's server will return
+    a response that is grounded to the provided list of passages. It will not
+    base the response on parametric memory.
+    """
+
+    _client: Any
+    _answer_style: int
+
+    def __init__(self, answer_style: int = 1, **kwargs: Any):
+        """Create a new Google AQA.
+
+        Args:
+          answer_style: See `google.ai.generativelanguage.AnswerStyle`
+        """
+        try:
+            import google.ai.generativelanguage as genai
+
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        super().__init__(
+            service_context=google_service_context,
+            output_cls=SynthesizedResponse,
+            **kwargs,
+        )
+        self._client = genaix.build_text_service()
+        self._answer_style = genai.AnswerStyle(answer_style)
+
+    def get_response(
+        self,
+        query_str: str,
+        text_chunks: Sequence[str],
+        **response_kwargs: Any,
+    ) -> SynthesizedResponse:
```

Comment:
You could also override `synthesize` and `asynthesize` to return a llama-index native response object

Something roughly like?
```
response = Response(
  synthesized_response.answer, 
  source_nodes=[
    SourceNode(
      node=TextNode(text="\n".join(synthesized_response.attributed_passages)), 
      score=synthesized_response.answerable_probability
    ),
  ]
) 
```

Reply:
Thanks for the suggestion! Added synthesize and asynthesize, although the async version is currently synchronous.

Reply:
That's fine, that sounds good, thanks!